Communication/src/com/gymfox/communication/Router.java

```diff
@@ -0,0 +1,52 @@
+package com.gymfox.communication;
+
+import java.util.Comparator;
+import java.util.TreeSet;
+
+public class Router {
+    private TreeSet<Route> routes = new TreeSet<Route>(new RouteComparator());
+
+    public Router(TreeSet<Route> newRoutes) throws Route.InvalidGatewayException {
+        defaultRoute();
+
+        this.routes.addAll(newRoutes);
+    }
+
+    private void defaultRoute() throws  Route.InvalidGatewayException {
+        addRoute(new Route(new Network(new IPv4Address("0.0.0.0"), 0), "en0", 10, new IPv4Address("0.0.0.1")));
+    }
+
+    public void addRoute(Route route) {
+        routes.add(route);
+    }
+
+    public Route getRouteForAddress(IPv4Address address) {
+        for ( Route route : routes ) {
+            if ( route.getNetworkAddress().contains(address) ) {
+                return route;
+            }
+        }
+
+        throw new RuntimeException();
+    }
+
+    public TreeSet<Route> getRoutes() {
+        return (TreeSet<Route>)routes.clone();
```

Comment:
It's not really good to use `clone()`.  Better to use coping construction: `return new TreeSet<Route>(routes)`

https://stackoverflow.com/a/2427946/4617634